internal/operators/ocs/validations.go

```diff
@@ -0,0 +1,209 @@
+package ocs
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/docker/go-units"
+	"github.com/openshift/assisted-service/internal/host"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -source=validations.go -package=ocs -destination=mock_validations.go
+type OcsValidator interface {
+	ValidateOCSRequirements(cluster *models.Cluster) bool
+}
+
+type ocsValidator struct {
+	*Config
+	log     logrus.FieldLogger
+	hostApi host.API
+}
+
+func NewOcsValidator(log logrus.FieldLogger, hostApi host.API, cfg *Config) OcsValidator {
+	return &ocsValidator{
+		log:     log,
+		Config:  cfg,
+		hostApi: hostApi,
+	}
+}
+
+type Config struct {
+	OCSMinimumCPUCount             int64 `envconfig:"OCS_MINIMUM_CPU_COUNT" default:"24"`
+	OCSMinimumRAMGB                int64 `envconfig:"OCS_MINIMUM_RAM_GB" default:"66"`
+	OCSRequiredDisk                int64 `envconfig:"OCS_MINIMUM_DISK" default:"3"` // hosts with disks must be a multiple of 3
+	OCSRequiredHosts               int64 `envconfig:"OCS_MINIMUM_HOST" default:"3"`
+	OCSRequiredCPUCount            int64 `envconfig:"OCS_REQUIRED_CPU_COUNT" default:"30"`
+	OCSRequiredRAMGB               int64 `envconfig:"OCS_REQUIRED_RAM_GB" default:"72"`
+	OCSRequiredCompactModeCPUCount int64 `envconfig:"OCS_REQUIRED_COMAPCT_MODE_CPU_COUNT" default:"12"` // compact mode cpu requirements ( 4(master)+8(OCS) cpus)
+	OCSRequiredCompactModeRAMGB    int64 `envconfig:"OCS_REQUIRED_COMAPCT_MODE_RAM_GB" default:"96"`    //compact mode ram requirements (24(master)+72(OCS) GB)
+	OCSMasterWorkerHosts           int   `envconfig:"OCS_REQUIRED_MASTER_WORKER_HOSTS" default:"5"`
```

Comment:
What does this indicate?

Reply:
Any better name suggestion for this? This was created to to fail the OCS validation for 3 masters, 2 workers.